newIDE/app/src/PropertiesEditor/PropertiesMapToSchema.js

```diff
@@ -27,11 +28,15 @@ export default (
       .getExtraInfo()
       .toJSArray()
       .map(value => ({ value, label: value }));
+    const dialog = PropertiesRenderingService.getDialogComponent(
+      property.getExtraInfo().toJSArray()[0]
```

Comment:
This looks risky to me. [0] could return undefined and you typed `getDialogComponent` as taking a string.

Reply:
Ah, yeah, I know, I feel a bit bad each time I read that "line", very WIP maybe?
I've to add some extra safe checks.

Reply:
Maybe you could actually move this into the `getDialog` function? Would defer the call to getDialogComponent until it's really needed - so no useless call and easier to debug calls to getDialogComponent too